state/constraints.go

```diff
@@ -115,19 +115,41 @@ func writeConstraints(mb modelBackend, id string, cons constraints.Value) error
 	return nil
 }
 
-// ConstraintsOpsForSpaceNameChange returns all the database transaction operation required
-// to transform a constraints spaces from `a` to `b`
-func (st *State) ConstraintsOpsForSpaceNameChange(from, to string) ([]txn.Op, error) {
+// ConstraintsTagForSpaceName returns the tags for the given space.
+func (st *State) ConstraintsTagForSpaceName(name string) ([]names.Tag, error) {
+	docs, err := st.constraintsBySpaceName(name)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	tags := make([]names.Tag, len(docs))
+	for i, doc := range docs {
+		tag := st.ParseLocalIDToTags(doc.DocID)
+		if tag == nil {
+			logger.Debugf("Could not parse id: %q", doc.DocID)
```

Comment:
Not sure whats the best way to solve this. 

- ignoring (just continue) it can lead to unexpected things
- but do we want to stop and just return here?

Reply:
Return an error (I don't think it will ever happen).

In any case, this method doesn't really belong in `state`. This is conversion for wire transport. Export `constraintsBySpaceName` and do this conversion at the API server.

Reply:
The reason why I initially did not export the type method was that it returns a slice of `docs` which we seem to not export.

I could either:
- export the type 
- create its own type, similar to constraints.Value, but just with an ID, from which I derive the tag.